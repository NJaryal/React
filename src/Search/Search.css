.searchBtn {
    background: #F65261;
    color: #fff;
    border-color: #F65261;
}