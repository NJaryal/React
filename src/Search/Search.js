import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { updateSearchText } from './SearchActions';
import Button from 'Components/common/Button';
import cx from 'classnames';
import globalStyles from '../../node_modules/bootstrap/dist/css/bootstrap.min.css';
import styles from './Search.css';

class Search extends React.Component {
  constructor(props) {
    super(props);
  }
  handleSearchTextChange = (e) => {
    this.props.updateSearchText(e.target.value);
  }
  render() {
    const  { inputs } = this.props;
    return (
      <form className={cx(globalStyles['navbar-form'], globalStyles['navbar-left'])} role="search">
        <div className={cx(globalStyles['input-group'], globalStyles['mb-3'])}>
          <input type="text" value={inputs} className={globalStyles['form-control']} placeholder="Enter movie's name" onChange={this.handleSearchTextChange} />
          <div className="input-group-append">
            <Button className={styles.searchBtn} children="Search" />
          </div>
        </div>
      </form>
    );
  }
}

Search.propTypes = {
  updateSearchText: PropTypes.func.isRequired,
  inputs: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  inputs: state.SearchReducer.inputs,
});

const mapDispatchToProps = dispatch => ({updateSearchText: searchText => {
  dispatch(updateSearchText(searchText));
},});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Search);
