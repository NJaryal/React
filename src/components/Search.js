import React from 'react'
import Button from './common/Button'
const Search = ({ resultsQuery, onKeyHandle }) => 
        <form className="navbar-form navbar-left" role="search">
            <div className="input-group mb-3">
                <input type="text" className="form-control" placeholder="Enter movie's name" onChange={onKeyHandle} />
                <div className="input-group-append">
                    <Button className="searchBtn" onClick={resultsQuery} children="Search"></Button>
                </div>
            </div>
        </form>

export default Search;



 