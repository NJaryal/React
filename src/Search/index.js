import React from 'react';
import PropTypes from 'prop-types';
import Button from 'Components/common/Button';
import './Search.css';

const Search = ({ resultsQuery, onKeyHandle }) => (
  <form className="navbar-form navbar-left" role="search">
    <div className="input-group mb-3">
      <input type="text" className="form-control" placeholder="Enter movie's name" onChange={onKeyHandle} />
      <div className="input-group-append">
        <Button className="searchBtn" onClick={resultsQuery} children="Search" />
      </div>
    </div>
  </form>
);
Search.propTypes = {
  resultsQuery: PropTypes.func.isRequired,
  onKeyHandle: PropTypes.func.isRequired,
};

export default Search;
