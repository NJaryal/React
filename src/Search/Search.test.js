import React from 'react';
import { shallow } from 'enzyme';
import { findByTestAttr } from '../../__tests__/Utils';
import { Search } from './Search';

const renderSearch = (props = {}) => {
  const component = shallow(<Search {...props} />);
  return component;
};

describe('should render Search', () => {
  let component;
  beforeEach(() => {
    component = renderSearch();
  });

  it('should render Search', () => {
    const findMovieText = findByTestAttr(component, 'SearchComponent');
    expect(findMovieText.length).toBe(1);
  });
});

describe('show', () => {
  let component;
  beforeEach(() => {
    component = renderSearch();
  });
  it('Snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
