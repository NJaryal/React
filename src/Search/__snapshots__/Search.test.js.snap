// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Snapshot 1`] = `
<form
  className="navbar-form navbar-left"
  data-test="SearchComponent"
  role="search"
>
  <div
    className="input-group mb-3"
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Enter movie's name"
      type="text"
    />
    <div
      className="input-group-append"
    >
      <Button
        className="searchBtn"
      >
        Search
      </Button>
    </div>
  </div>
</form>
`;

exports[`Search component show Snapshot 1`] = `
<form
  className="navbar-form navbar-left"
  data-test="SearchComponent"
  role="search"
>
  <div
    className="input-group mb-3"
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Enter movie's name"
      type="text"
    />
    <div
      className="input-group-append"
    >
      <Button
        className="searchBtn"
      >
        Search
      </Button>
    </div>
  </div>
</form>
`;

exports[`show Snapshot 1`] = `
<form
  className="navbar-form navbar-left"
  data-test="SearchComponent"
  role="search"
>
  <div
    className="input-group mb-3"
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Enter movie's name"
      type="text"
    />
    <div
      className="input-group-append"
    >
      <Button
        className="searchBtn"
      >
        Search
      </Button>
    </div>
  </div>
</form>
`;
